src/content/get-started/data-and-json.md

```diff
@@ -0,0 +1,481 @@
+      static Summary fromJson(Map<String, Object?> json) {
+        return switch (json) {
+          {
+            'titles': final Map<String, Object?> titles,
+            'pageid': final int pageid,
+            'extract': final String extract,
+            'extract_html': final String extractHtml,
+            'lang': final String lang,
+            'dir': final String dir,
+            'content_urls': {
+              'desktop': {'page': final String url},
+              'mobile': {'page': String _},
+            },
+            'description': final String description,
+          } =>
+            Summary(
+              titles: TitlesSet.fromJson(titles),
+              pageid: pageid,
+              extract: extract,
+              extractHtml: extractHtml,
+              lang: lang,
+              dir: dir,
+              url: url,
+              description: description,
+            ),
+          {
+            'titles': final Map<String, Object?> titles,
+            'pageid': final int pageid,
+            'extract': final String extract,
+            'extract_html': final String extractHtml,
+            'lang': final String lang,
+            'dir': final String dir,
+            'content_urls': {
+              'desktop': {'page': final String url},
+              'mobile': {'page': String _},
+            },
+          } =>
+            Summary(
+              titles: TitlesSet.fromJson(titles),
+              pageid: pageid,
+              extract: extract,
+              extractHtml: extractHtml,
+              lang: lang,
+              dir: dir,
+              url: url,
+            ),
+          _ => throw FormatException('Could not deserialize Summary, json=$json'),
+        };
+      }
+
+      @override
+      String toString() =>
+          'Summary['
+          'titles=$titles, '
+          'pageid=$pageid, '
+          'extract=$extract, '
+          'extractHtml=$extractHtml, '
+          'lang=$lang, '
+          'dir=$dir, '
+          'description=$description'
+          ']';
+    }
+    ```
+
+    This code defines a `Summary` class with properties that correspond to the
+    fields in the JSON response from the Wikipedia API.
+    The `fromJson` method uses [pattern matching][] to
+    extract the data from the JSON object and create a new `Summary` instance.
+    The `toString` method provides a convenient way to
+    print the contents of the `Summary` object.
+    Note that the `TitlesSet` class is used in the `Summary` class,
+    so you'll need to create that next.
+
+[pattern matching]: /language/patterns
+
+### Task 4: Create the TitleSet class
+
+The `Summary` class uses a `TitlesSet` class to represent the title information.
+Let's create that class now.
+
+1.  Create the file `wikipedia/lib/src/model/title_set.dart`.
+
+1.  Add the following code to `wikipedia/lib/src/model/title_set.dart`:
+
+    ```dart title="wikipedia/lib/src/model/title_set.dart"
+    class TitlesSet {
```

Comment:
I'd consider a different name than `TitlesSet`, it makes me think it contains multiple different titles and/or implements the `Set` interface.

It seems it contains alternative versions of a single title. So perhaps just `ArticleTitle`  or `TitleInfo`?

\cc @lamek @ericwindmill